openedx/core/lib/api/permissions.py

```diff
@@ -80,3 +82,31 @@ def has_object_permission(self, request, view, obj):
         return (request.user.is_staff or
                 CourseStaffRole(obj.course_id).has_user(request.user) or
                 request.method in permissions.SAFE_METHODS)
+
+
+def is_field_shared_factory(field_name):
```

Comment:
@andy-armstrong and I talked this over-- we feel that the best thing is to move this permissions class into the user_api package itself, since it is specific to user_api (whereas this core api package is meant to hold more generic classes). You can create a permissions.py file within user_api/accounts to hold it. Hopefully that will break the circular dependency issue.